automation-roles/99-generic/openshift/openshift-login/tasks/aws-login-rosa-ocp.yml

```diff
@@ -28,10 +28,19 @@
     name: vault-get-secret
   vars:
     secret_name: "{{ _p_openshift_cluster_name }}-cluster-admin-password"
-    secret_group: "{{ environment_name }}" 
+    secret_group: "{{ environment_name }}"
+  failed_when: vault_get_secret.rc != 0
```

Comment:
Please confirm that you tested this. So far I know the vault-get-secret role does not return a rc element. To validate whether the secret was existing, you have to check that the secret_value is blank.

Reply:
I added a modification checks if secret_value is defined and not empty (secret_value != '') before setting the _rosa_cluster_admin_password fact. This ensures that the fact is only set if a non-blank value was retrieved from the vault.